recipes_source/mobile_perf.rst

```diff
@@ -0,0 +1,191 @@
+Pytorch Mobile Performance Recipes
+==================================
+
+Introduction
+------------
+Performance (aka latency) is crucial to most, if not all,
+applications and use-cases of ML model inference on mobile devices.
+
+Today, PyTorch executes the models on the CPU backend pending availability
+of other hardware backends such as GPU, DSP, and NPU.
+
+Model preparation
+-----------------
+
+Next recipes you can take (offline) while preparing the model
+to have an optimized model that will probably have shorter execution time
+(higher performance, lower latency) on the mobile device.
+
+1. Use torch.utils.mobile_optimizer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Torch mobile_optimizer package does several optimizations with the model,
+which will help to conv2d and linear operations.
+It pre-packs model weights in an optimized format and fuses ops above with relu
+if it is the next operation.
+
+::
+
+  from torch.utils.mobile_optimizer import optimize_for_mobile
+  traced_model = torch.jit.load("input_model_path")
+  optimized_model = optimize_for_mobile(traced_model)
+  torch.jit.save(optimized_model, "output_model_path")
+
+2. Fuse operators using ``torch.quantization.fuse_modules``
+Do not be confused that fuse_modules is in the quantization package.
+It works for all types of torch script modules.
```

Comment:
Two things:
1) PMM is going to come back and say that we should write it TorchScript.
2) The code below *does not* pass a TorchScript module to `fuse_modules()` - that MobileNet v2 from TorchVision is not a subclass of `ScriptModule`.

Passing in the original torchvision module itself or a version of it processed by `torch.jit.script()` works, with the latter giving ~2% file size improvement.